### PR TITLE
Ahmedkhalifa/dropdownrefactor

### PR DESCRIPTION
**Pull request by Khalifa1997**
Added DropDown Component which utilizes props.children and props as seen below, also for some reason couldn't target the Link component and add .list-item class to it, so used inline styles instead.

### CODE REVIEW

File: src/components/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  children: any;
+  href: string;
+  passedDownStyles: string;
+  onClick?(): void;
+}
+type IProps = IOwnProps;
+
+const Dropdown = ({ children, href, onClick, passedDownStyles }: IProps) => {
+  return (
+    <LinkContainer>
```
Comment (Ananasship) [inline src/components/Dropdown.tsx]: Should this be named something like `DropdownContainer`?

File: src/components/NavMenu.tsx [hunk 2]
```diff
@@ -165,12 +109,70 @@ class NavMenu extends React.Component<IProps, IState> {
       'dataset',
       'contact',
     ];
-    const links = linksFactory({
+    /* const links = linksFactory({
```
Comment (Ananasship) [inline src/components/NavMenu.tsx]: Nit: remove comment

File: src/components/NavMenu.tsx [hunk 3]
```diff
@@ -165,12 +109,70 @@ class NavMenu extends React.Component<IProps, IState> {
...
       profile: this.props.profile,
-    });
+    }); */
+    console.log(this.props);
```
Comment (Ananasship) [inline src/components/NavMenu.tsx]: Nit: remove console.log

File: src/components/NavMenu.tsx [hunk 4]
```diff
@@ -192,11 +194,26 @@ class NavMenu extends React.Component<IProps, IState> {
...
-                  }
-                )}
+                {dropDownList.map((item, index) => {
+                  const classNames = classnames({
+                    active: item.href === this.props.location.pathname,
+                    busy: item.busy,
+                    badge: item.badgeText,
+                  });
+                  return (
+                    <Dropdown
```
Comment (Ananasship) [inline src/components/NavMenu.tsx]: Is this meant to be a link or a dropdown? 

Dropdown is meant to be the whole container, link is a single item.

File: src/components/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import classnames from 'classnames';
+import { withRouter, Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { ILink } from './NavMenu';
+import T from './T';
+
+interface IOwnProps {
+  children: any;
+  href: string;
```
[the remark this answers is not in the document]
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: The dropdown component resembles an element in the dropdown, thus what come to my head is no matter what the object in the dropdown is, it has to have a link that navigates the user somewhere, this the href

File: src/components/Dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  passedDownStyles: string;
+  onClick?(): void;
```
[the remark this answers is not in the document]
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: Based on the ILink interface, some ILink objects have an onclick, I didn't want to change the base configuration of how you did things so thats why I added it there

File: src/components/Dropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import { withRouter, Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { ILink } from './NavMenu';
+import T from './T';
+
+interface IOwnProps {
+  children: any;
+  href: string;
+  passedDownStyles: string;
```
[the remark this answers is not in the document]
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: This one was a tricky one, probably there is much better ways to using this more effectively, I imagine one was with styled-components, but the reason why I passed it down, is cause I want to target the `Link` component in `Dropdown` not the whole thing

File: src/components/Dropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    <LinkContainer>
+      <Link
+        to={href}
+        onClick={() => {
+          if (onClick) {
+            onClick();
+          }
+        }}
+        className={passedDownStyles}
+        style={{
```
[the remark this answers is not in the document]
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: I couldn't figure out how to do it with `styled-components` so this was just a temporary thing till I learn `styled-components`, which I will in the next few days.

File: src/components/Dropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+}
+type IProps = IOwnProps;
+
+const Dropdown = ({ children, href, onClick, passedDownStyles }: IProps) => {
+  return (
+    <LinkContainer>
+      <Link
+        to={href}
+        onClick={() => {
+          if (onClick) {
```
[the remark this answers is not in the document]
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: This one I just used from the old configuration and couldn't really guess why it was used like this, I think that check is pointless however

File: src/components/Dropdown.tsx (hunk 1 above)
Comment (Khalifa1997) [reply src/components/Dropdown.tsx]: Yes, that would of been a more convenient name.

File: src/components/NavMenu.tsx (hunk 4 above)
Comment (Khalifa1997) [reply src/components/NavMenu.tsx]: The current config is a single dropdown item, but having a dropdown container would make more sense, I can change this now

**Pull request closed without merging**